tests/unit/lastuser_core/test_model_user_User.py

```diff
@@ -0,0 +1,466 @@
+        self.assertIsInstance(crusoe.phone, models.UserPhone)
+        self.assertEqual(crusoe_phone, crusoe.phone)
+        self.assertTrue(crusoe.phone.primary)
+        # scenario 2: when there is a phone but not as primary
+        snowball = models.User(username=u'snowball')
+        snowball_phone = models.UserPhone(phone=u'+918574808032', user=snowball)
+        db.session.add_all([snowball, snowball_phone])
+        db.session.commit()
+        self.assertIsInstance(snowball.phone, models.UserPhone)
+        self.assertEqual(snowball_phone, snowball.phone)
+        self.assertTrue(snowball.phone.primary)
+        # scenario 3: when there is no phone on db
+        piglet = self.fixtures.piglet
+        assert piglet.phone is u''
+
+    def test_user__set_password(self):
+        """
+        Test to set user password
+        """
+        # Scenario 1: Set None as password
+        castle = models.User(username=u'castle', fullname=u'Rick Castle')
+        self.assertEqual(castle._set_password(None), None)
+        # Scenario 2: Set valid password
+        kate = models.User(username=u'kate', fullname=u'Detective Kate Beckette')
+        kate_password = kate._set_password(u'12thprecinct')
+        db.session.add(kate)
+        db.session.commit()
+        result = models.User.get(userid=kate.userid)
+        self.assertEqual(len(result.pw_hash), 60)
+        self.assertGreater(result.pw_expires_at, result.pw_set_at)
+
+    def test_user_clients_with_team_access(self):
+        """
+        Test to verify a list of clients with access to the user's organizations' teams.
+        """
+        aro = models.User(username=u'aro')
+        jane = models.User(username=u'jane')
+        marcus = models.User(username=u'marcus')
+        volturi = models.Organization(name=u'volturi', title=u'The Volturi')
+        volturi.owners.users.append(aro)
+        volturi.members.users.append(marcus)
+        volturi.members.users.append(jane)
+        volturi.make_teams()
+        volterra = models.Client(title=u'Volterra, Tuscany', org=volturi, confidential=True, website=u'volterra.co.it')
+        enforcers = models.Client(title=u'Volturi\'s thugs', org=volturi, confidential=True, website=u'volturi.co.it')
+        volterra_auth_token = models.AuthToken(client=volterra, user=aro, scope=u'teams', validity=0)
+        enforcers_auth_token = models.AuthToken(client=enforcers, user=marcus, scope=u'teams', validity=0)
+        self.assertItemsEqual(aro.clients_with_team_access(), [volterra])
+        self.assertItemsEqual(marcus.clients_with_team_access(), [enforcers])
+        self.assertEqual(jane.clients_with_team_access(), [])
+
+    def test_user_password_has_expired(self):
+        """
+        Test to check if password for a user has expired
+        """
+        alexis = models.User(username=u'alexis', fullname=u'Alexis Castle')
+        alexis._set_password(u'unfortunateincidents')
+        alexis.pw_expires_at=datetime.utcnow()+ timedelta(0,0,1)
```

Comment:
`password_has_expired()` just does a check with the current time, so it's safe to user a negative delta, which is more reliable.